### PR TITLE
chore: migrate from ansiterm to nu_ansi_term

### PR DESCRIPTION
**Pull request by PThorpe92**
`nu_ansi_term` has a few features we could use including built-in [hyperlinks](https://docs.rs/nu-ansi-term/latest/nu_ansi_term/struct.AnsiGenericString.html#method.hyperlink) and [color-gradients](https://docs.rs/nu-ansi-term/latest/nu_ansi_term/gradient/struct.Gradient.html) that could move some of the work we currently do in `eza`.

This PR just migrates us to the dependency, it doesn't move any features over just yet.

### CODE REVIEW

**Comment by MartinFillon** [comment]
gonna take a look tomorrow or later tonight but this is a great advancement

**Comment by PThorpe92** [comment]
Forgot to mention, the reason this didn't work out last attempt was that part of `color-scale` had to get re-implemented. I honestly thought it might break more stuff, but initial testing looked fine. The `color-scale` will have to be tested more thoroughly as I am not 100% on my fix.

**Comment by MartinFillon** [comment]
gonna wait for the tests to pass

**Comment by PThorpe92** [comment]
wtf.. I had handled a merge conflict in the GH ui, but it must not have been fully committed or something because i pulled `main` into the branch and resolved it locally now all the tests pass. weird

**Comment by MartinFillon** [comment]
> I had handled a merge conflict in the GH ui

Well such a simple error, never use gh ui xd

File: src/main.rs [hunk 1]
```diff
@@ -53,11 +53,6 @@ fn main() {
 
     logger::configure(env::var_os(vars::EZA_DEBUG).or_else(|| env::var_os(vars::EXA_DEBUG)));
 
-    #[cfg(windows)]
-    if let Err(e) = ansiterm::enable_ansi_support() {
-        warn!("Failed to enable ANSI support: {}", e);
-    }
-
```
Comment (MartinFillon) [inline src/main.rs:60]: nice loosing this boilerplate

**Review by MartinFillon — DISMISSED** [review]
This is a great start when merged it will surely open many options

**Comment by PThorpe92** [comment]
I just realized we are going to have to enable feature flag for windows. Gonna set it up for compilation dependent :+1:

**Comment by PThorpe92** [comment]
Config file is almost ready for PR. It will be dependent on `nu_ansi_term` so we should think about getting this merged pretty soon :+1:

**Review by cafkafk — CHANGES_REQUESTED** [review]
Can you resolve the conflicts, I'll review after :+1:

**Comment by cafkafk** [comment]
Superseded by #910

**Pull request closed without merging**